doc/adr/0015-current-API-authentication-strategy.md

```diff
@@ -0,0 +1,80 @@
+# 0015. A description of the current API authentication strategy
+
+## Status
+
+Accepted
+
+## Problem
+
+We have no documentation around how authentication works in Weave Gitops. This will aim to capture the current state of
+the system.
+
+The 'decision' section has been skipped as the decisions have been made and reverse-engineering the rationale for them is unlikely to work. The 'context' section covers implementation details and the 'consequences' aims to enumerate major risks with the system as well as how to work with it safely.
+
+This supersedes the sections of [ADR-0005 Wego Core Auth Strategy](./0005-wego-core-auth-strategy.md) not to do with authentication to GitHub/GitLab.
```

Comment:
@foot Are we sure this is true? Do we authenticate with any git providers these days? We left that code hanging around in Core because EE was relying on it, if I remember correctly.

Reply:
Given that this doc is in the Core repo, I guess it is technically true that we don't do this in Core anymore.